src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -635,64 +635,76 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
+                // transfer and build the uri in the response with a status of Image Available
+
+                // If the protocol requested is not supported, return status - Protocol Not Supported
+                if (!isBDXProtocolSupported) {
+                    Commands::QueryImageResponse::Type response;
+                    response.status
+                        = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                    handler->AddResponse(cachedCommandPath, response);
+                    handle.Release();
+                    return;
+                }
 
-                    char uriBuffer[kMaxBDXURILen];
-                    MutableCharSpan uri(uriBuffer);
-                    err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
+                Commands::QueryImageResponse::Type delegateResponse;
+                ConvertFromQueryImageResponseParms(data, delegateResponse);
+
+                // If update is not available, return the delegate response
+                if (!hasUpdate) {
+                    handler->AddResponse(cachedCommandPath, delegateResponse);
+                    handle.Release();
+                    return;
+                }
+
+                // If there is an update available, return a response with status - Available and set the image Uri in the
+                // response
+                auto fabricIndex = handler->GetSubjectDescriptor().fabricIndex;
+                auto nodeId = handler->GetSubjectDescriptor().subject;
+                CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
+                if (CHIP_NO_ERROR != err) {
+
+                    // Handle busy error separately as we have a query image response status that maps to busy
+                    if (err == CHIP_ERROR_BUSY) {
+                        ChipLogError(Controller, "Resource is busy handling another request.");
+                        Commands::QueryImageResponse::Type response;
+                        response.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
+                        response.delayedActionTime.SetValue(delegateResponse.delayedActionTime.ValueOr(kDelayedActionTimeSeconds));
+                        handler->AddResponse(cachedCommandPath, response);
                         handle.Release();
-                        gOtaSender.ResetState();
                         return;
                     }
-
-                    response.imageURI.SetValue(uri);
-                    handler->AddResponse(cachedCommandPath, response);
+                    LogErrorOnFailure(err);
+                    handler->AddStatus(cachedCommandPath, StatusIB(err).mStatus);
```

Comment:
This is fine, but now I am very confused by this comment in `ConfigureState`:
```
        // The caller of this method maps CHIP_ERROR to specific BDX Status Codes (see GetBdxStatusCodeFromChipError)
        // and those are used by the BDX session to prepare the status report.
```
This code is the only way `ConfigureState` is reached, right?  And we're not calling GetBdxStatusCodeFromChipError here....

Reply:
I think we do that for aborttransfer not prepare transfer anymore. I need to modify the comment accordingly

Reply:
This should just return the err now. fixed it.